### PR TITLE
[FEAT] Add similarity search support from the SDK

### LINKED ISSUE #4934: [FEAT] Querying records using vectors
`dataset.find_simlars` ?

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

```python
dataset.records(
    query=rg.Query(
        similar=rg.Similar("my_vector", (0.1, 0.2, 0.3)),
    ),
)

record = dataset.records(...)

dataset.records(
    query=rg.Query(
        similar=rg.Similar("my_vector", record),
    ),
)
```

Closes #4934 

**Type of change**

- Improvement (change adding some improvement to an existing functionality)
- Documentation update

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: argilla-server/src/argilla_server/api/handlers/v1/datasets/records.py [hunk 1]
```diff
@@ -167,13 +168,20 @@ async def _get_search_responses(
             "record": record,
             "query": text_query,
             "order": vector_query.order,
-            "max_results": limit,
+            "max_results": min(limit + offset, SEARCH_MAX_SIMILARITY_SEARCH_RESULT),
```
Comment (burtenshaw) [inline argilla-server/src/argilla_server/api/handlers/v1/datasets/records.py:171]: Based on this name, won't the user expect to just limit similarity but actually they're limiting all queries?

File: argilla/src/argilla/_models/_search.py [hunk 2]
```diff
@@ -97,10 +116,22 @@ class TextQueryModel(BaseModel):
     field: Union[str, None] = None
 
 
+class VectorQueryModel(BaseModel):
+    name: str
+    record_id: Optional[UUID] = None
+    value: Optional[List[float]] = None
+    order: Literal["most_similar", "least_similar"] = "most_similar"
```
Comment (burtenshaw) [inline argilla/src/argilla/_models/_search.py:123]: Do you think we could abstract this in the SDK to a bool?

**Review by burtenshaw — COMMENTED** [review]
In general looks good, just two comments to improve the naming.

File: argilla-server/src/argilla_server/api/handlers/v1/datasets/records.py (hunk 1 above)
Comment (frascuchon) [reply argilla-server/src/argilla_server/api/handlers/v1/datasets/records.py:171]: This argument affects only the similarity searches. Search without victory similarity does not have this limitation.

File: argilla/src/argilla/_models/_search.py (hunk 2 above)
Comment (frascuchon) [reply argilla/src/argilla/_models/_search.py:123]: Something like: 

```python
rg.Similar(most_similar=False)
rg.Similar(least_similar=True)
```

File: argilla-server/src/argilla_server/api/handlers/v1/datasets/records.py (hunk 1 above)
Comment (burtenshaw) [reply argilla-server/src/argilla_server/api/handlers/v1/datasets/records.py:171]: Got it.

File: argilla/src/argilla/_models/_search.py (hunk 2 above)
Comment (burtenshaw) [reply argilla/src/argilla/_models/_search.py:123]: That works. Do we need both?

Comment (frascuchon) [reply argilla/src/argilla/_models/_search.py:123]: Nope. I will add the `most_similar` flag

**Review by burtenshaw — APPROVED** [review]

File: argilla/tests/integration/test_search_records.py [hunk 3]
```diff
@@ -12,10 +12,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from dataclasses import fields
 from datetime import datetime
+from random import random
 
 import pytest
+from mpmath import limit
```
Comment (frascuchon) [inline argilla/tests/integration/test_search_records.py]:
```suggestion
```

**Pull request merged**